### PR TITLE
Handle failure when streaming

### PR DESCRIPTION
**Pull request by chulkilee**
Fixes #48

- [x] Use real error messages
- [x] Fix credo warnings ;)

### CODE REVIEW

File: lib/ex_force.ex [hunk 1]
```diff
@@ -350,34 +352,44 @@ defmodule ExForce do
 
   @spec start_query_stream(
           client,
-          (client, soql -> {:ok, QueryResult.t()} | any),
+          (client, soql -> {:ok, QueryResult.t()} | {:error, any}),
```
Comment (suranyami) [inline lib/ex_force.ex:355]: I see… so the Stream is always returned, but the query doesn't happen until the first item is requested, potentially returning an error tuple.

Very nice!

File: test/ex_force_test.exs [hunk 2]
```diff
@@ -844,6 +844,72 @@ defmodule ExForceTest do
...
+          }
+        ]
+      }
+      """)
+    end)
+
+    Bypass.expect_once(bypass, "GET", "/services/data/v40.0/query/queryid-200", fn conn ->
+      conn
+      |> Conn.put_resp_content_type("application/json")
+      |> Conn.resp(500, """
```
Comment (suranyami) [inline test/ex_force_test.exs:897]: Nice!

**Review by suranyami — APPROVED** [review]
Top work!

**Comment by suranyami** [comment]
So, I'm guessing, I should close #48 ?

**Comment by suranyami** [comment]
@chulkilee So, did this not get merged into `main` in the end? Why?

**Comment by chulkilee** [comment]
Wait, I thought I merged it and pushed to `main` 🤦  let me fix that

**Pull request closed without merging**